boromir674/cookiecutter-python-package
# Cookiecutter Python Package v1.5.0 Release

1.5.0 (2022-06-11)
==================

This release focused on improving the code architecture, reducing technical
debt, decoupling components, cleaning code, fixing styling issues.

It also features some updates in the Generated Project, with improved tox envs,
cleaner python scripts and cleaner development tools' configuration files
(such as .pylintrc, pyproject.toml, tox.ini).

Changes
-------

feature
"""""""
- document config settings, improve tox envs & scripts

test
""""
- verify Generator can be invoked as python module: `python -m cookiecutter_python`

refactor
""""""""
- dry code per string_sanitizer implementation
- lint, type check & style code
- reduce code in cli.py by delegating error handling to the new cli_handlers.py module
- reduce code of parse_version.py script
- abstract input sanitization
- reduce main code
- import 3 objects in pre_gen_project from backend, instead of 4
- decouple components
- decouple code
- clean code
- clean code, satisfy some todos, dry code

ci
""
- run prospector as part of the Lint Step in the Job
- use the template's parse_version script to reduce duplicate code
- enable test workflow for prs to 'develop' branch and simplify build matrix quick feedback

chore
"""""
- enable full build matrix for test workflow
